src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs

```diff
@@ -32,42 +33,59 @@ public class ResourceIdentifier
 
         public ResourceIdentifier() { }
 
+        private class UrlTagName
+        {
+            public const string subscriptions = "subscriptions",
+                resourceGroups = "resourceGroups",
+                providers = "providers";
+        }
+
         public ResourceIdentifier(string idFromServer)
         {
             if (!string.IsNullOrEmpty(idFromServer))
             {
                 string[] tokens = idFromServer.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
-                if (tokens.Length < 8)
+                if (tokens.Length % 2 > 0 || tokens.Length < 8)
```

Comment:
This assumption may not correct. below are 2 examples
/subscriptions/{}/resourcegroup/{}/providers/{}/resourceType/{resource}/{operation}
/subscriptions/{}/resourcegroup/{}/providers/{}/resourceType/{resource}/subresources